---
title: "RWire"
author: "Fabian"
---

## Introduction
RWire is an R package for the correlation analysis of single-cell ATAC-seq data. This vignette shows how to use it to calculate an accessibility matrix and a correlation matrix, which contain information on co-regulated genomic regions. For further information see the 'sample-scripts' folder.

## Installation
RWire can be installed by starting R and typing:
```
install.packages("devtools")
devtools::install_github("FabianErdel/RWire")
```

## Loading libraries
The following libraries are required to run the analysis:
```
library("ggplot2")
library("data.table")
library("GenomicRanges")

library("RWire")
```

## Making a list of genomic regions of interest
With the following function, 1 kb-tiles for chromosome 18 are generated, which can be used with the sample data provided with this package.
```
tiles <- makeTiles(18)
```

## Loading sample data
The sample data provided with this package contain counts from 100 human B cells in the genomic region surrounding the TCF4 gene on chromosome 18.
```
path <- system.file("extdata", package = "RWire")
```

## Making an accessibility matrix
```
# make accessibility matrix from a set of bed files
am <- makeAccMatrix(path, tiles)

# delete empty rows
am <- am[rowSums(am[,4:dim(am)[2]])>0, ]
```

## Retrieving some statistics
The following code creates histograms for the number of integrations per cell and for the 'number of cells with integrations' per genomic region, and retrieves some basic information about the distributions. This information can be useful to detect outliers and to assess the overall data quality.
```
hist_ipc <- makeIPCHist(am)
print(hist_ipc)
makeIPCStats(am)

hist_ipr <- makeIPRHist(am)
print(hist_ipr[[1]])
makeIPRStats(am)
```

## Making a correlation matrix
This function creates a correlation matrix, which contains a Pearson correlation coefficient for each pair-wise combination of genomic regions. Entry (*i*, *j*) contains the correlation coefficient for regions *i* and *j*.
```
cor <- makeCorMatrix(am)
```

## Making a scaled correlation matrix
This function makes a correlation matrix with genomic positions on its axes, i.e., entry (*i*, *j*) contains the average of the pair-wise correlation coefficients for the regions in genomic bin *i* and genomic bin *j*.
```
size <- 200
scm <- scaleCorMatrix(cor, size)
```

## Saving the correlation matrix as an image
This function makes a graphical representation of the correlation matrix.
```
saveImage(scm, "your_path", size, size, getCorLUT(), -1, 1)
```
