---
title: "RWire"
author: "Fabian"
---

## Introduction
RWire is an R package for the correlation analysis of single-cell ATAC-seq data. This vignette shows how to use it to calculate accessibility matrices and correlation matrices in order to identify co-regulated genomic regions.

## Installation
RWire can be installed by starting R and typing:
```
install.packages("devtools")
devtools::install_github("FabianErdel/RWire")
```

## Loading required libraries
```
library("ggplot2")
library("data.table")
library("GenomicRanges")
```

## Making a list of genomic regions of interest
With the following function, 1 kb-tiles for chromosome 18 are generated, which can be used with the sample data provided with this package 
```
tiles <- makeTiles(18)
```

## Loading sample data
```
path <- system.file("extdata", package = "RWire")
```

## Making an accessibility matrix
```
# make accessibility matrix from a set of bed files
am <- makeAccMatrix(path, tiles)

# delete empty rows
am <- am[rowSums(am[,4:dim(am)[2]])>0, ]
```

## Retrieving some statistics
The following code creates a histogram for the number of integrations per cell and for the number of integrations per region. Such histograms can be useful to detect outliers and to assess the overall quality of the data.
```
hist_ipc <- makeIPCHist(am)
print(hist_ipc)
makeIPCStats(am)

hist_ipr <- makeIPRHist(am)
print(hist_ipr[[1]])
makeIPRStats(am)
```

## Making a correlation matrix
This function makes a correlation matrix, which contains a Pearson correlation coefficient for each pair-wise combination of genomic regions. Entry (*i*, *j*) contains the correlation coefficient for regions *i* and *j*.
```
cor <- makeCorMatrix(am)
```

## Making a scaled correlation matrix
This function makes a correlation matrix with genomic positions on its axes, i.e., entry (*i*, *j*) contains the (average) correlation coefficient for regions in genomic bin *i* and genomic bin *j*.
```
size <- 200
scm <- scaleCorMatrix(cor, size)
```

## Transforming the correlation matrix into an image
```
saveImage(scm, "your_path", size, size, getCorLUT(), -1, 1)
```
